client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -356,6 +357,16 @@ public void flushPartitions(ChannelPoolPartitionSelector selector) {
         }
     }
 
+    @Override
+    public long getIdleChannelCount() {
+        return partitions.reduceValuesToLong(
+                Long.MAX_VALUE,
```

Comment:
Since this parameter decides how many elements are needed to activate parallelism, `Long.MAX_VALUE` pretty much switches off parallel reduction of the elements. Do you think that's too conservative?